pkg/generator_test.go

```diff
@@ -702,6 +707,127 @@ func (s *GeneratorSuite) TestGeneratorVariadicArgs() {
 	)
 }
 
+func (s *GeneratorSuite) TestGeneratorVariadicArgsAsOneArg() {
+	// Read the expected output from a "golden" file that we can also import in CompatSuite
+	// to asserts its actual behavior.
+	//
+	// To regenerate the golden file: make fixture
+	expectedBytes, err := ioutil.ReadFile(filepath.Join(fixturePath, "mocks", "requester_variadic_one_arg.go"))
+	s.NoError(err)
+	expected := string(expectedBytes)
+	expected = expected[strings.Index(expected, "// RequesterVariadicOneArgument is"):]
+	generator := NewGenerator(
+		s.ctx, config.Config{
+			StructName: "RequesterVariadicOneArgument",
+			InPackage:  true,
+			Vararg:     true,
+		}, s.getInterfaceFromFile("requester_variadic.go", "RequesterVariadic"), pkg,
+	)
+	s.NoError(generator.Generate(s.ctx), "The generator ran without errors.")
+
+	var actual []byte
+	actual, fmtErr := format.Source(generator.buf.Bytes())
+	s.NoError(fmtErr, "The formatter ran without errors.")
+
+	expectedLines := strings.Split(expected, "\n")
+	actualLines := strings.Split(string(actual), "\n")
+
+	s.Equal(
+		expectedLines, actualLines,
+		"The generator produced unexpected output.",
+	)
+}
+
+func TestRequesterVariadicOneArgument(t *testing.T) {
```

Comment:
I've added tests to check if testify works correctly with the new mocks. Please let me know if I better delete them.